### PR TITLE
[DI] Don't generate factories for errored services

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

Spotted while comparing the dumped container of Blackfire with Symfony 4.0 vs 4.1.

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php [hunk 1]
```diff
@@ -1318,6 +1323,18 @@ private function exportParameters(array $parameters, string $path = '', int $ind
 
     private function endClass(): string
     {
+        if ($this->addThrow) {
+            return <<<'EOF'
+
+    protected function throw($message)
```
Comment (Tobion) [inline src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: I don't see the advantage of creating a method for this.

File: src/Symfony/Component/Finder/phpunit.xml.dist [hunk 2]
```diff
@@ -1,30 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-
-<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (Tobion) [inline src/Symfony/Component/Finder/phpunit.xml.dist]: Why remove this file?

Comment (nicolas-grekas) [reply src/Symfony/Component/Finder/phpunit.xml.dist]: a mistake, fixed

File: src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: We need to throw inline, ie in expressions, but the `throw` keyword is not allowed there.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**